detox/test/scripts/postinstall.js

```diff
@@ -1,29 +1,66 @@
+const fs = require('fs-extra');
+const path = require('path');
+const semver = require('semver');
+
 const rnVersion = function() {
   const rnPackageJson = require('react-native/package.json');
   return rnPackageJson.version;
 }();
 
 function patchHermesLocationForRN60Android() {
-  const semver = require('semver');
-  const fs = require('fs-extra');
-  const path = require('path');
+  const HERMES_PATH_ROOT = path.join('node_modules', 'hermesvm');
+  const HERMES_PATH_RN = path.join('node_modules', 'react-native', 'node_modules', 'hermesvm');
 
-  if (semver.minor(rnVersion) === 60) {
-    console.log('Detox post-install: Detected RN .60...');
+  const hermesIsInRoot = fs.existsSync(HERMES_PATH_ROOT);
+  const hermesIsInRN = fs.existsSync(HERMES_PATH_RN);
 
-    const HERMES_PATH_ROOT = path.join('node_modules', 'hermesvm');
-    const HERMES_PATH_RN = path.join('node_modules', 'react-native', 'node_modules', 'hermesvm');
+  if (hermesIsInRoot && !hermesIsInRN) {
+    console.log('  Applying hermes-vm patch for RN .60...');
+    fs.ensureDirSync(path.join(HERMES_PATH_RN, 'android'));
+    fs.copySync(path.join(HERMES_PATH_ROOT, 'android'), path.join(HERMES_PATH_RN, 'android'));
+  } else {
+    console.log('  Skipping hermes-vm patching (not needed):', hermesIsInRoot, hermesIsInRN);
+  }
+}
 
-    const hermesIsInRoot = fs.existsSync(HERMES_PATH_ROOT);
-    const hermesIsInRN = fs.existsSync(HERMES_PATH_RN);
+/**
+ * In RN .64, it seems that react-native-codegen - a proprietary code generation plugin, whose
+ * native code is only available inside react-native's monorepo, has been applied.
+ * This patch disables it, and that works - as long as RN is used as an .aar dep and
+ * not built from source. I can't account for the latter; It is likely that with .64 it may not
+ * be possible to do anymore. If ever needed, one approach to try out is to somehow build the
+ * plugin ourselves through the react-native monorepo itself (onto a .jar), and then add it as a
+ * direct plugin dependency into our settings.gradle (https://docs.gradle.org/current/userguide/plugins.html#sec:custom_plugin_repositories).
+ */
+function overrideReactAndroidGradleForRn64Android() {
```

Comment:
Sounds like an interesting story, @d4vidi. I'd be happy if we schedule a brief talk 1:1 so that you show me more context in person.

Reply:
Hey np!